### PR TITLE
Reduce regex matching to improve performance

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

**Comment by pbek** [comment]
Really looks like a great idea!

**Comment by Waqar144** [comment]
I have tried to keep the contains rules  as unique as possible. You should take a look at them again(especially the link and image rules) to be sure

File: markdownhighlighter.cpp [hunk 1]
```diff
@@ -115,6 +115,9 @@ void MarkdownHighlighter::initHighlightingRules() {
     {
         HighlightingRule rule(HighlighterState::MaskedSyntax);
         rule.pattern = QRegularExpression(QStringLiteral(R"(^\[.+?\]: \w+://.+$)"));
+        rule.shouldContain[0] = QStringLiteral("://");
+        rule.shouldContain[1] = QLatin1String("");
```
Comment (pbek) [inline markdownhighlighter.cpp]: do you really need to set empty strings if you don't need them? in my tests these lines also could be let out

File: markdownhighlighter.cpp [hunk 2]
```diff
@@ -1678,6 +1762,20 @@ void MarkdownHighlighter::highlightAdditionalRules(
                 continue;
             }
 
+            bool contains = false;
+            if (text.contains(rule.shouldContain[0])) {
```
Comment (pbek) [inline markdownhighlighter.cpp]: I guess all those ifs can be accumulated in one if with some "or" operators :laughing:

File: markdownhighlighter.cpp (hunk 1 above)
Comment (Waqar144) [reply markdownhighlighter.cpp]: I am trying another way. It is faster. And it doesn't contain these.

They aren't necessary though.

**Pull request merged**